Common/Data/Consolidators/VolumeRenkoConsolidator.cs

```diff
@@ -0,0 +1,143 @@
+    public class VolumeRenkoConsolidator : DataConsolidator<BaseData>
+    {
+        private bool _firstTick = true;
+        private VolumeRenkoBar _currentBar;
+        private decimal _volumeLeftOver = 0m;
+        private decimal _barSize;
+
+        /// <summary>
+        /// Gets a clone of the data being currently consolidated
+        /// </summary>
+        public override IBaseData WorkingData => _currentBar;
+
+        /// <summary>
+        /// Gets <see cref="VolumeRenkoBar"/> which is the type emitted in the <see cref="IDataConsolidator.DataConsolidated"/> event.
+        /// </summary>
+        public override Type OutputType => typeof(VolumeRenkoBar);
+
+        /// <summary>
+        /// Gets the most recently consolidated piece of data. This will be null if this consolidator
+        /// has not produced any data yet.
+        /// </summary>
+        public IBaseData Consolidated
+        {
+            get; private set;
+        }
+
+        /// <summary>
+        /// Event handler that fires when a new piece of data is produced
+        /// </summary>
+        public event EventHandler<VolumeRenkoBar> DataConsolidated;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="VolumeRenkoConsolidator"/> class using the specified <paramref name="barSize"/>.
+        /// </summary>
+        /// <param name="barSize">The constant volume size of each bar</param>
+        public VolumeRenkoConsolidator(decimal barSize)
+        {
+            _barSize = barSize;
+        }
+
+        /// <summary>
+        /// Updates this consolidator with the specified data
+        /// </summary>
+        /// <param name="data">The new data for the consolidator</param>
+        public override void Update(BaseData data)
+        {
+            var rate = data.Price;
+            var dataType = data.GetType();
+
+            decimal volume;
+            decimal open;
+            decimal high;
+            decimal low;
+            decimal close;
+
+            if (dataType == typeof(TradeBar))
+            {
+                volume = ((TradeBar)data).Volume;
+                open = ((TradeBar)data).Open;
+                high = ((TradeBar)data).High;
+                low = ((TradeBar)data).Low;
+                close = rate;
+            }
+            else if (dataType == typeof(Tick))
+            {
+                // Only include actual trade information
+                if (((Tick)data).TickType != TickType.Trade)
+                {
+                    return;
+                }
+
+                volume = ((Tick)data).Quantity;
+                open = rate;
+                high = rate;
+                low = rate;
+                close = rate;
+            }
+            else
+            {
+                throw new ArgumentException("VolumeRenkoConsolidator() must be used with TradeBar or Tick data.");
+            }
+
+            if (_firstTick)
+            {
+                _firstTick = false;
```

Comment:
- Can avoid the need for this flag just by checking if `_currentBar` is null or not?
- What if the first bar already covers the expected volume? we should emit right away, seems we are waiting for the next data point
- Seems `_volumeLeftOver` can just be a local variable?